packages/vite/src/client/overlay.ts

```diff
@@ -44,10 +44,11 @@ const template = /*html*/ `
 .window {
   font-family: var(--monospace);
   line-height: 1.5;
-  width: 800px;
+  max-width: 80vw;
```

Comment:
```suggestion
  width: 80vw;
  max-width: 800px;
```

How about doing `max-width: 800px;` with `width: 80vw;`?
I believe that `max-width` should be a static value.
If we use `max-width: 80vw;`, it would expand infinitely and I believe it looks bad.
This applies to `padding` as well.

I am bringing up an extreme environment where screen width is `4000px` wide, but I think you will get the point.

[`visual viewport` based]
<img width="1544" alt="Screenshot 2024-02-28 at 16 34 38" src="https://github.com/vitejs/vite/assets/48273875/20646671-612e-4c16-acac-775bb265e585">

[`max-width: 800px` w/ `px` padding]
<img width="1544" alt="Screenshot 2024-02-28 at 16 35 42" src="https://github.com/vitejs/vite/assets/48273875/8c2b7d2e-0fc6-4fe0-aecc-aea633a5c5dc">

Reply:
Now that I see here, I think it would be better to have the block margin of `.window` to be dynamic to viewport height, i.e. `margin: 10vh auto`

Reply:
the problem is that, the `overlay` error info is not designed for Flexible screen,  if U use big screen, the main problem maybe the font-size;

Reply:
the problem is that, the `overlay` error info is not designed for Flexible screen,  if U use big screen, the main problem maybe the font-size;

Reply:
I get what you mean, but as we are introducing this change to support mobile viewport, we should consider the wider viewport as well.

If you think my opinion is unrelated to your PR, I will open another PR after this one being merged and continue the discussion! 😊